### PR TITLE
Update TestApp to React Native 0.58.3

### PR DESCRIPTION
**Pull request by roland-chernov-akvelon**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR updates TestApp to use React Native 0.58.3.

## Misc

- `react-native-modal-selector` doesn't support React Native 0.58 (peacechen/react-native-modal-selector#99). Fixed version from a fork fix is used.
- `.babelrc` was replaced by `babel.config.js` ([diff](https://github.com/pvinis/rn-diff-purge/compare/version/0.57.3...version/0.58.3)).
- `android.useDeprecatedNdk=true` property is deprecated ([diff](https://github.com/facebook/react-native/commit/6117a6c7205c969f93d39ba02e0583881572d5fa#diff-975c8e77f36db951fb304aa3d843b2fd)).

### CODE REVIEW

File: TestApp/android/gradle.properties [hunk 1]
```diff
@@ -16,5 +16,3 @@
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
-
-android.useDeprecatedNdk=true
```
Comment (dhei) [inline TestApp/android/gradle.properties]: @roland-chernov-akvelon Do we still need this or it's safe to get rid of it?
@guperrot Any comments?

File: TestApp/android/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,5 +1,3 @@
 <resources>
-    <string name="appCenterCrashes_whenToSendCrashes" moduleConfig="true">ASK_JAVASCRIPT</string>
```
Comment (dhei) [inline TestApp/android/app/src/main/res/values/strings.xml]: Please keep these strings.

File: TestApp/android/app/src/main/java/com/testapp/MainApplication.java [hunk 3]
```diff
@@ -1,58 +1,45 @@
 package com.testapp;
 
 import android.app.Application;
-import android.util.Log;
```
Comment (dhei) [inline TestApp/android/app/src/main/java/com/testapp/MainApplication.java]: Probably you shouldn't change anything in this file.

File: TestApp/android/app/proguard-rules.pro [hunk 4]
```diff
@@ -15,52 +15,3 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
-
```
Comment (dhei) [inline TestApp/android/app/proguard-rules.pro]: Is the change in this file intentional? I think we should keep it.

File: TestApp/.babelrc [hunk 5]
```diff
@@ -1,3 +0,0 @@
-{
```
Comment (dhei) [inline TestApp/.babelrc]: What is the purpose of this file change?

Comment (dhei) [reply TestApp/.babelrc]: Let me answer my own question - it's a change introduced by RN 0.58. (see the diff of blank new RN app at https://github.com/pvinis/rn-diff-purge/compare/version/0.57.8...version/0.58.0)

**Comment by dhei** [comment]
After reading RN docs, I think a practical way to upgrade RN version would be first upgrade the react-native dependency, then upgrade your project templates as documented by [Upgrading to new React Native versions
](https://facebook.github.io/react-native/docs/upgrading#alternative).

Please take a look of [RN 0.58 release notes](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#058):

> Please note that upgrading iOS users, you'll need to manually link JavaScriptCore.framework; this is done here in XCode.

> Additionally for upgrading Android users, please note that Android's target SDK 27 is supported. Work is still underway to land target SDK 28 support, and it will come soon.

We want to be sure that we're not ignoring breaking changes that affects our SDK.

File: TestApp/android/gradle.properties (hunk 1 above)
Comment (roland-chernov-akvelon) [reply TestApp/android/gradle.properties]: There were a lot of changes generated by the update utility. We kept only necessary and reverted this file back. Please let me know if we can remove this line.

File: TestApp/android/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (roland-chernov-akvelon) [reply TestApp/android/app/src/main/res/values/strings.xml]: Sure, sorry. These changes were generated by the update utility. We kept only necessary changes.

File: TestApp/.babelrc (hunk 5 above)
Comment (roland-chernov-akvelon) [reply TestApp/.babelrc]: Yes, you're right. Thank you for the link, it's very helpful.

File: TestApp/android/gradle.properties (hunk 1 above)
Comment (dhei) [reply TestApp/android/gradle.properties]: See [RN v0.57.3 release notes](https://github.com/facebook/react-native/commit/6117a6c7205c969f93d39ba02e0583881572d5fa#diff-975c8e77f36db951fb304aa3d843b2fd)

> bump Android NDK to r17c (b7d0594 by @dulmandakh)

and [diff which removes `android.useDeprecatedNdk=true`](https://github.com/facebook/react-native/commit/6117a6c7205c969f93d39ba02e0583881572d5fa#diff-975c8e77f36db951fb304aa3d843b2fd).

Comment (roland-chernov-akvelon) [reply TestApp/android/gradle.properties]: Thank you, removed.

Comment (dhei) [reply TestApp/android/gradle.properties]: We don't support NDK in RN SDK yet so it should be safe to remove this line.

File: TestApp/ios/TestApp/AppDelegate.m [hunk 6]
```diff
@@ -38,7 +38,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   
   [AppCenterReactNativePush register];  // Initialize AppCenter push
 
-  [AppCenterReactNativeCrashes register];  // Initialize AppCenter crashes
```
Comment (dhei) [inline TestApp/ios/TestApp/AppDelegate.m]: Please revert this.

**Comment by peacechen** [comment]
@roland-chernov-akvelon 
The RN 0.58 fix for `react-native-modal-selector` has been published on NPM as v1.0.2.

**Comment by roland-chernov-akvelon** [comment]
@peacechen Wow, great! Thank you.

File: TestApp/ios/TestApp/AppDelegate.m (hunk 6 above)
Comment (roland-chernov-akvelon) [reply TestApp/ios/TestApp/AppDelegate.m]: Fixed, thank you!

**Pull request merged**